mmseg/models/backbones/hrformer.py

```diff
@@ -0,0 +1,1071 @@
+                self.window_size[0] * self.window_size[1],
+                self.window_size[0] * self.window_size[1],
+                -1,
+            )  # Wh*Ww,Wh*Ww,nH
+        relative_position_bias = relative_position_bias.permute(
+            2, 0, 1).contiguous()  # nH, Wh*Ww, Wh*Ww
+        attn_output_weights = attn_output_weights.view(
+            bsz, num_heads, tgt_len,
+            src_len) + relative_position_bias.unsqueeze(0)
+        attn_output_weights = attn_output_weights.view(bsz * num_heads,
+                                                       tgt_len, src_len)
+        """
+        Reweight the attention map before softmax().
+        attn_output_weights: (b*n_head, n, hw)
+        """
+        attn_output_weights = softmax(attn_output_weights, dim=-1)
+        attn_output_weights = dropout(
+            attn_output_weights, p=dropout_p, training=training)
+
+        attn_output = torch.bmm(attn_output_weights, v)
+        assert list(
+            attn_output.size()) == [bsz * num_heads, tgt_len, v_head_dim]
+        attn_output = (
+            attn_output.transpose(0,
+                                  1).contiguous().view(tgt_len, bsz, out_dim))
+        attn_output = linear(attn_output, self.out_proj.weight,
+                             self.out_proj.bias)
+        return attn_output
+
+
+class LocalWindowSelfAttention(nn.Module):
+    r""" Local-window Self Attention (LSA) module with relative position bias.
+
+    This module is the short-range self-attention module in the
+    Interlaced Sparse Self-Attention <https://arxiv.org/abs/1907.12273>`_.
+
+    Args:
+        window_size (tuple[int]): Window size.
+    """
+
+    def __init__(self, *args, window_size=7, **kwargs):
+        super(LocalWindowSelfAttention, self).__init__()
+
+        self.window_size = window_size
+        self.attn = MultiheadAttention(
+            *args, window_size=window_size, **kwargs)
+
+    def forward(self, x, H, W, **kwargs):
+        B, N, C = x.shape
+        x = x.view(B, H, W, C)
+        Wh = Ww = self.window_size
+
+        # center-pad the feature on H and W axes
+        pad_h = math.ceil(H / Wh) * Wh - H
+        pad_w = math.ceil(W / Ww) * Ww - W
+        x = pad(x, (0, 0, pad_w // 2, pad_w - pad_w // 2, pad_h // 2,
+                    pad_h - pad_h // 2))
+
+        # permute
+        x = x.view(B, math.ceil(H / Wh), Wh, math.ceil(W / Ww), Ww, C)
+        x = x.permute(2, 4, 0, 1, 3, 5)
+        x = x.reshape(Wh * Ww, -1, C)
+
+        # attention
+        out = self.attn(x, x, x, **kwargs)
+
+        # reverse permutation
+        out = out.reshape(Wh, Ww, B, math.ceil(H / Wh), math.ceil(W / Ww), C)
+        out = out.permute(2, 0, 3, 1, 4, 5)
+        out = out.reshape(B, H + pad_h, W + pad_w, C)
+
+        # de-pad
+        out = out[:, pad_h // 2:H + pad_h // 2, pad_w // 2:W + pad_w // 2]
+        return out.reshape(B, N, C)
+
+
+class CrossFFN(nn.Module):
```

Comment:
Can we reuse  `MixFFN` in [`mit`](https://github.com/open-mmlab/mmsegmentation/blob/master/mmseg/models/backbones/mit.py#L18) as that of `LocallyGroupedSelfAttention` in `twins`?